packages/calypso-e2e/src/lib/blocks/block-flows/contact-form.ts

```diff
@@ -0,0 +1,58 @@
+import { BlockFlow, EditorContext, PublishedPostContext } from '..';
+
+interface ConfigurationData {
+	text: string;
+}
+
+const blockParentSelector = 'div[aria-label="Block: Form"]';
+const selectors = {
+	// Editor
+	nameBlock: 'div[aria-label="Block: Name"]',
+
+	// Published
+	publishedBlock: '.wp-block-jetpack-contact-form',
+	nameField: 'input[class="name"]',
+};
+
+/**
+ * Class representing the flow of using an block in the editor.
+ */
+export class ContactFormFlow implements BlockFlow {
+	private configurationData: ConfigurationData;
+
+	/**
+	 * Constructs an instance of this block flow with data to be used when configuring and validating the block.
+	 *
+	 * @param {ConfigurationData} configurationData data with which to configure and validate the block
+	 */
+	constructor( configurationData: ConfigurationData ) {
+		this.configurationData = configurationData;
+	}
+
+	blockSidebarName = 'Contact Form';
+	blockEditorSelector = blockParentSelector;
+
+	/**
+	 * Configure the block in the editor with the configuration data from the constructor
+	 *
+	 * @param {EditorContext} context The current context for the editor at the point of test execution
+	 */
+	async configure( context: EditorContext ): Promise< void > {
+		// Add a new paragraph block after the Name block and insert some text.
+		await context.editorIframe.click( selectors.nameBlock );
+		await context.page.keyboard.press( 'Enter' );
+		await context.page.keyboard.insertText( this.configurationData.text );
```

Comment:
Personally, I don't think adding the paragraph block between form pieces adds a lot of utility to the smoke test. If anything, we could maybe change the title of a field? I think though that with the smoke of scope testing, and given Contact Forms basically come pre-configured, we could also just make the configure function a no-op!

Reply:
I agree that it doesn't add much to the test. 

I chose this step because I've found the forms to be quite tricky to interact with, and I didn't want to introduce any potential environment related issues with the idea to edit the field title. 

In a hypothetical test, something like the following was imagined:
- click on 'Email' field.
- select all.
- delete.
- fill in with other text.

The (potential) issue I ran into with the above approach was the `select all` portion. It is unclear whether the shortcut keys supported by Playwright is cross-platform compatible.

Examples taken from [Playwright documentation](https://playwright.dev/docs/api/class-keyboard):
```
// on Windows and Linux
await page.keyboard.press('Control+A');
// on macOS
await page.keyboard.press('Meta+A');
```

I suppose an alternative that I hadn't explored was to manipulate the label directly, with something like:

```
document.getElementById('Label1').innerHTML = ''
```

Reply:
Update: it wasn't as complicated as I had initially thought. 

Turns out Playwright's `page.fill( selector, '')` ends up clearing the label too!

Reply:
Updated in 6b7814f2e6ad4ffab1646d2e940327bb6745bffc.